target/linux/ath79/dts/qca9558_comfast_wr650ac.dts

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca9557.dtsi"
+
+/ {
+	compatible = "comfast,cf-wr650ac-v1", "qca,qca9557";
+	model = "Comfast CF-WR650AC v1";
+
+	chosen {
+                bootargs = "console=ttyS0,115200n8";
+        };
+
+	leds {
+		compatible = "gpio-leds";
+
+		wps {
+			label = "cf-wr650ac:blue:wps";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+		};
+
+		network {
+			label = "cf-wr650ac:blue:network";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "cf-wr650ac:blue:wlan2g";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan5g {
+			label = "comfast:blue:wlan5g";
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		button0 {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+};
+
+&pinmux {
+	pinmux_pa_dcdc_pins {
+		pinctrl-single,bits = <0x0 0xff00 0x0>;
+	};
+
+	pinmux_pa_high_pins {
+		pinctrl-single,bits = <0x10 0xff 0x0>;
+	};
+};
+
+&pcie0 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x020000>;
+				read-only;
+			};
+
+			partition@1 {
+				label = "art";
+				reg = <0x020000 0x010000>;
+				read-only;
+			};
+
+			partition@2 {
+                                label = "firmware";
+                                reg = <0x030000 0xfc0000>;
+                        };
+
+			partition@3 {
+				label = "art-backup";
+				reg = <0xff0000 0x010000>;
+				read-only;
+			}
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+		phy-mode = "rgmii-id";
+	};
+};
+
+&mdio1 {
+	status = "okay";
+
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+		phy-mode = "sgmii";
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	pll-data = <0xa6000000>;
+
+	phy-handle = <&phy0>;
+	phy-mode = "rgmii";
+};
+
+&eth1 {
+	status = "okay";
+
+	pll-data = <0x03000101>;
```

Comment:
@grosjo this is probably wrong, you should provide settings for 1000/100/10 speeds.

Reply:
/* GMAC0 is connected to the RMGII interface */
	ath79_eth0_data.phy_if_mode = PHY_INTERFACE_MODE_RGMII;
	ath79_eth0_data.phy_mask = CF_WR650AC_LAN_PHYMASK;
	ath79_eth0_data.mii_bus_dev = &ath79_mdio0_device.dev;
	ath79_eth0_pll_data.pll_1000 = 0xa6000000;
	ath79_register_eth(0);

	/* GMAC1 is connected to the SGMII interface */
	ath79_eth1_data.phy_if_mode = PHY_INTERFACE_MODE_SGMII;
	ath79_eth1_data.speed = SPEED_1000;
	ath79_eth1_data.duplex = DUPLEX_FULL;
	ath79_eth1_pll_data.pll_1000 = 0x03000101;
	ath79_register_eth(1);

Reply:
Indeed, your value for 1000 is probably correct, but it won't work for others as the driver expects array of 3 values `<1000 100 10>`:

https://github.com/openwrt/openwrt/blob/13c379e5c6e37b3822bf6e4e18cc6317e5b40fbb/target/linux/ath79/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_main.c#L1383-L1384

for example `pll-data = <0xa6000000 0x00000101 0x00001616>;`

You can do `git grep pll-data` in OpenWrt source directory to check it yourself.

Reply:
@ynezz vendor says "it is setup by default at uboot time", so I commented the line

Reply:
This is probably wrong approach, please provide proper `pll-data` values as done in ar71xx. Values from bootloader are going to work until first soft reset, where the PLL register values would be set to defaults.

Reply:
I put the 10 and 100 values of similar qca9558-based boards:

pll-data = <0xa6000000 0xB0000101 0xB0001313>;

(the 1000 value is the one from vendor)